TheApiAwakens/StarWarsDetailViewController.swift

```diff
@@ -0,0 +1,551 @@
+        
+        switch type {
+        case .starship:
+            let starships = getSmallestAndlargest(from: self.starships)
+            smallestLabelValue.text = starships.smallest.name
+            largestLabelValue.text = starships.largest.name
+        case .vehicle:
+            let vehicle = getSmallestAndlargest(from: vehicles)
+            smallestLabelValue.text = vehicle.smallest.name
+            largestLabelValue.text = vehicle.largest.name
+        case .character, .none: break
+        }
+        
+        // Only transportCraft(Vehcile and Starship have costLabel)
+        self.valueSelectedTransportCraft = valueSelected
+        // All the types have sizeLabel
+        self.valueSelectedAllTypes = valueSelected
+        
+        nameValueLabel.text = valueSelected.name
+        makeValueLabel.text = valueSelected.make
+        costValueLabel.text = String(valueSelected.cost.cleanValue)
+        crewValueLabel.text = valueSelected.crew
+        classValueLabel.text = valueSelected.swClass
+        
+        if valueSelected.size == 0 {
+            lengthValueLabel.text = "Unknown"
+        } else {
+            lengthValueLabel.text = String(valueSelected.size.cleanValue)
+        }
+    }
+
+    // Defines who is the smallest and largest
+    func getSmallestAndlargest<T: Measurable>(from resource: [T]) -> (smallest: T, largest: T) {
+        // DOn't use sizes which the value is = 0. 0 means that the size is unknown
+        let resourceFiltered = resource.filter { (Measurable) -> Bool in
+            if Measurable.size != 0 {
+                return true
+            } else {
+                return false
+            }
+        }
+        
+        let largest = resourceFiltered.max { a, b in a.size < b.size }
+        let smallest = resourceFiltered.min { a, b in a.size < b.size }
+        return (smallest: smallest!, largest: largest!)
+    }
+
+    func fetchForCharacter(with page: Int) {
+        swAPIClient.fetchForCharacter(nextPage: page, completion: { [weak self] (result) in
+            switch result {
+            case .failure(let error):
+                print(error)
+            case .success(let result):
+                self?.hasFinishedRequest = true
+                
+                // Update pickerView with data
+                self?.pickerData = [String]()
+                self?.characters += result.resource
+                self?.objectQuantity = (self?.characters.count)!
+                self?.hasNextPage = result.hasPage
+
+                for character in (self?.characters)! {
+                    self?.pickerData.append(character.name)
+                }
+                self?.pickerView.reloadAllComponents()
+                
+                //Set labels with first element of the array since the user has not selected any row on the pickerView yet
+                let firstCharacter = self?.characters.first!
+                if (self?.isApiFirstCall)! {
```

Comment:
It seems that all the code inside this if is duplicated in `func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {`